File: public/docs/_examples/dependency-injection/ts/app/car/car.ts
// #docregion
import {Injectable} from 'angular2/core';

// #docregion engine
export class Engine {
  public cylinders = 4; // default
}
// #enddocregion engine

// #docregion tires
export class Tires {
  public make  = 'Flintstone';
  public model = 'Square';
}
// #enddocregion tires

@Injectable()
// #docregion car
export class Car {
  //#docregion car-ctor
  public description = 'DI';

  constructor(public engine: Engine, public tires: Tires) { }
  // #enddocregion car-ctor

  // Method using the engine and tires
  drive() {
    return `${this.description} car with ` +
      `${this.engine.cylinders} cylinders and ${this.tires.make} tires.`
  }
}
// #enddocregion car

File: public/docs/_examples/dependency-injection/dart/lib/car/car.dart
import 'package:angular2/core.dart';

@Injectable()
class Engine {
  final int cylinders;

  Engine() : cylinders = 4;
  Engine.withCylinders(this.cylinders);
}

@Injectable()
class Tires {
  String make = 'Flintstone';
  String model = 'Square';
}

@Injectable()
class Car {
  //#docregion car-ctor
  final Engine engine;
  final Tires tires;
  String description = 'DI';

  Car(this.engine, this.tires);
  // #enddocregion car-ctor

  // Method using the engine and tires
  String drive() => '$description car with '
      '${engine.cylinders} cylinders and '
      '${tires.make} tires.';
}
